File: server/src/main/java/vn/edu/actvn/server/entity/Attendance.java
package vn.edu.actvn.server.entity;

import com.fasterxml.jackson.annotation.JsonCreator;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
@Table(name = "attendance")
public class Attendance {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    String attendanceId;

    @ManyToOne
    @JoinColumn(name = "class_id")
    EntityClass entityClass;

    @Column(nullable = false)
    LocalDate date;

    @ElementCollection
    @CollectionTable(name = "attendance_student", joinColumns = @JoinColumn(name = "attendance_id"))
    List<StudentAttendance> studentAttendances;

    public enum Status {
        PRESENT, ABSENT
    }
}

File: client/src/types/attendance.ts

export interface AttendanceUpdateRequest {
    date: string;
    studentAttendances: StudentAttendance[];
}

export interface StudentAttendance {
    studentId: string;
    status: string;
    note: string;
}

export interface AttendanceResponse {
    attendanceId: string;
    classId: string;
    date: string;
    studentAttendances: StudentAttendance[];
}
